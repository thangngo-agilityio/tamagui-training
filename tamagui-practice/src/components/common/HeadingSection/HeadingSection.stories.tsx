import { Meta, StoryObj } from '@storybook/react';

// Component
import HeadingSection from '.';

const meta: Meta<typeof HeadingSection> = {
  title: 'Common/Heading',
  tags: ['autodocs'],
  component: HeadingSection,

  parameters: {
    controls: {
      expanded: true,
    },
  },
};

export default meta;

type Story = StoryObj<typeof HeadingSection>;

export const Default: Story = {
  args: {
    title: 'Title',
  },
};
