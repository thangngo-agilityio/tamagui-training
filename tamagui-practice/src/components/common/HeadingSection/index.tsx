import { Flex, Heading } from '@chakra-ui/react';

type THeadingSection = {
  title: string;
};

const HeadingSection = ({ title }: THeadingSection) => (
  <Flex
    w="100%"
    flexDir="column"
    borderBottom="1px solid #DEE5EF"
    mt="65px"
    mb="35px"
  >
    <Heading size="size5xl" variant="quinary" mb="8px">
      {title}
    </Heading>
  </Flex>
);

export default HeadingSection;
