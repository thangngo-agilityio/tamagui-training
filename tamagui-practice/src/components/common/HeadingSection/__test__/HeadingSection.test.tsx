import React from 'react';
import { render } from '@testing-library/react';
import HeadingSection from '../index';
import '@testing-library/jest-dom';

describe('HeadingSection() HeadingSection method', () => {
  const title = 'Test Title';
  it('should match snapshot for HeadingSection', () => {
    const { container } = render(<HeadingSection title={title} />);

    expect(container).toMatchSnapshot();
  });

  describe('Happy Path', () => {
    it('should render the HeadingSection with the provided title', () => {
      // Act
      const { getByText } = render(<HeadingSection title={title} />);

      // Assert
      expect(getByText(title)).toBeInTheDocument();
    });

    it('should apply the correct styles to the Flex container', () => {
      // Arrange
      const title = 'Styled Title';

      // Act
      const { container } = render(<HeadingSection title={title} />);
      const flexElement = container.firstChild;

      // Assert
      expect(flexElement).toHaveStyle('width: 100%');
      expect(flexElement).toHaveStyle('flex-direction: column');
      expect(flexElement).toHaveStyle('border-bottom: 1px solid #DEE5EF');
      expect(flexElement).toHaveStyle('margin-top: 65px');
      expect(flexElement).toHaveStyle('margin-bottom: 35px');
    });
  });

  // Edge Case Tests
  describe('Edge Cases', () => {
    it('should handle a very long title without crashing', () => {
      // Arrange
      const longTitle = 'A'.repeat(1000); // A very long string

      // Act
      const { getByText } = render(<HeadingSection title={longTitle} />);

      // Assert
      expect(getByText(longTitle)).toBeInTheDocument();
    });
  });
});
