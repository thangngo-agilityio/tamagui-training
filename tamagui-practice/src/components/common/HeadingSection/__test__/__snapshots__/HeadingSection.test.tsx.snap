// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadingSection() HeadingSection method should match snapshot for HeadingSection 1`] = `
<div>
  <div
    class="css-icp0c2"
  >
    <h2
      class="chakra-heading css-19idom"
    >
      Test Title
    </h2>
  </div>
</div>
`;
